pkg/apis/metrics/v1alpha1/types.go

```diff
@@ -16,14 +16,208 @@ limitations under the License.
+}
+
+// AggregateSample contains a metric sample point of data aggregated across containers.
+type AggregateSample struct {
+	Sample `json:",inline"`
+	// Metrics pertaining to CPU resources.
+	CPU *CPUMetrics `json:"cpu,omitempty"`
+	// Metrics pertaining to memory (RAM) resources.
+	Memory *MemoryMetrics `json:"memory,omitempty"`
+	// Metrics pertaining to network resources.
+	Network *NetworkMetrics `json:"network,omitempty"`
+}
+
+// PodSample contains a metric sample point of pod-level resources.
+type PodSample struct {
+	Sample `json:",inline"`
+	// Metrics pertaining to network resources.
+	Network *NetworkMetrics `json:"network,omitempty"`
+}
+
+// ContainerSample contains a metric sample point of container-level resources.
+type ContainerSample struct {
+	Sample `json:",inline"`
+	// Metrics pertaining to CPU resources.
+	CPU *CPUMetrics `json:"cpu,omitempty"`
+	// Metrics pertaining to memory (RAM) resources.
+	Memory *MemoryMetrics `json:"memory,omitempty"`
+}
+
+// NetworkMetrics contains data about network resources.
+type NetworkMetrics struct {
+	// Cumulative count of bytes received.
+	RxBytes *resource.Quantity `json:"rxBytes,omitempty"`
+	// Cumulative count of receive errors encountered.
+	RxErrors *int64 `json:"rxErrors,omitempty"`
+	// Cumulative count of bytes transmitted.
+	TxBytes *resource.Quantity `json:"txBytes,omitempty"`
+	// Cumulative count of transmit errors encountered.
+	TxErrors *int64 `json:"txErrors,omitempty"`
+}
+
+// CPUMetrics contains data about CPU usage.
+type CPUMetrics struct {
+	// Total CPU usage (sum of all cores) averaged over the sample window.
+	// The "core" unit can be interpreted as CPU core-seconds per second.
+	TotalCores *resource.Quantity `json:"totalCores,omitempty"`
+	// CPU load that the container is experiencing, represented as a smoothed average of number of
+	// runnable threads.  Load is averaged over the sampling window.
+	LoadAverage *resource.Quantity `json:"loadAverage,omitempty"`
+}
+
+// MemoryMetrics contains data about memory usage.
+type MemoryMetrics struct {
+	// Total memory in use. This includes all memory regardless of when it was accessed.
+	TotalBytes *resource.Quantity `json:"totalBytes,omitempty"`
+	// The amount of working set memory. This includes recently accessed memory,
+	// dirty memory, and kernel memory. UsageBytes is <= TotalBytes.
+	UsageBytes *resource.Quantity `json:"usageBytes,omitempty"`
+	// Cumulative number of minor page faults.
+	PageFaults *int64 `json:"pageFaults,omitempty"`
+	// Cumulative number of major page faults.
+	MajorPageFaults *int64 `json:"majorPageFaults,omitempty"`
+}
+
+// CustomMetricType specifies the type of metric being exported.
+type CustomMetricType string
+
+const (
+	// Instantaneous value. May increase or decrease.
+	CustomMetricGauge CustomMetricType = "Gauge"
+	// A counter-like value that increases monotonically.
+	CustomMetricCumulative CustomMetricType = "Cumulative"
```

Comment:
I don't have an opinion either way on this issue. If as Jimmi says the prometheus names are becoming standard, it may make sense to rename the cAdvisor types to match (in a V3 API, if we ever do that).